app/database/server/models/post.ts

```diff
@@ -0,0 +1,110 @@
+ */
+export default class Post extends Model {
+    /** table (entity name) : Post */
+    static table = POST;
+
+    /** associations : Describes every relationship to this entity. */
+    static associations: Associations = {
+
+        /** A CHANNEL can have multiple POST. (relationship is 1:N) */
+        [CHANNEL]: {type: 'belongs_to', key: 'channel_id'},
+
+        /** A POST can have multiple DRAFT. (relationship is 1:N) */
+        [DRAFT]: {type: 'has_many', foreignKey: 'root_id'},
+
+        /** A POST can have multiple FILE.  (relationship is 1:N)*/
+        [FILE]: {type: 'has_many', foreignKey: 'post_id'},
+
+        /** A POST can have multiple POSTS_IN_THREAD. (relationship is 1:N)*/
+        [POSTS_IN_THREAD]: {type: 'has_many', foreignKey: 'post_id'},
+
+        /** A POST can have multiple POST_METADATA. (relationship is 1:N)*/
+        [POST_METADATA]: {type: 'has_many', foreignKey: 'post_id'},
+
+        /** A POST can have multiple REACTION. (relationship is 1:N)*/
+        [REACTION]: {type: 'has_many', foreignKey: 'post_id'},
+
+        /** A USER can have multiple POST.  A user can author several posts. (relationship is 1:N)*/
+        [USER]: {type: 'belongs_to', key: 'user_id'},
+    };
+
+    /** channel_id : The foreign key for the Channel to which this post belongs to. */
+    @field('channel_id') channelId!: string;
+
+    /** create_at : The timestamp to when this post was first created */
+    @field('create_at') createAt!: number;
+
+    /** delete_at : The timestamp to when this post was last archived/deleted */
+    @field('delete_at') deleteAt!: number;
+
+    /** edit_at : The timestamp to when this post was last edited */
+    @field('edit_at') editAt!: number;
+
+    /** is_pinned : A Boolean flag indicating if this Post is pinned */
+    @field('is_pinned') isPinned!: boolean;
+
+    /** message : Message in the post */
+    @field('message') message!: string;
+
+    /** original_id : Any post will have this value empty unless it is updated */
+    @field('original_id') originalId!: string;
+
+    /** pending_post_id : The id given to a post before it is published on the server */
+    @field('pending_post_id') pendingPostId!: string;
+
+    /** previous_post_id : Id of the previous post.  If this value is empty, this implies that it is not in the db and we will request it from server */
+    @field('previous_post_id') previousPostId!: string;
+
+    /** root_id : Used in threads. All posts under a thread will have this id in common */
+    @field('root_id') rootId!: string;
+
+    /** type : Type of props (e.g. system message) */
+    @field('type') type!: string;
+
+    /** user_id : The foreign key of the User who authored this post. */
+    @field('user_id') userId!: string;
+
+    /** props : Additional attributes for this props */
+    @json('props', (rawJson) => rawJson) props!: string;
+
+    /** drafts  : Every drafts associated with this Post */
+    @children(DRAFT) drafts!: Draft;
```

Comment:
this is wrong, there can be only one draft associated with this post as long as this post is a parent post

Reply:
@migbot  You are right. It should be an array there.

@enahum Please confirm the following : 

```
A Post should have a 1:1 relationship with Draft.

A Channel should/can have a 1:N relationship with Draft.

```

I will rewrite the updates in this direction

Reply:
Corrected